src/components/LoginForm.jsx

```diff
@@ -0,0 +1,52 @@
+import { Link } from 'react-router-dom';
+
+const LoginForm = ({ login, loading, error }) => {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const username = event.target[0].value;
+    const password = event.target[1].value;
+    login(username, password);
+  };
+
+  if (loading) {
+    return <div className='container'>Loading...</div>;
+  }
+  return (
+    <div>
+      {error && <div className='alert alert-danger'>{error.msg}</div>}
+      <div className='col-md-8 col-lg-6 col-xl-4 offset-xl-1 mx-auto pt-5'>
+        <h2 className="mb-4">Log In</h2>
+        <form className="mb-3" onSubmit={handleSubmit}>
+          <div className='mb-3'>
+            <label htmlFor='username' className='form-label'>
+              Username
+            </label>
+            <input
```

Comment:
don't you think we need to make these https://reactjs.org/docs/forms.html#controlled-components